src/replication/log_consumer.hpp

```diff
@@ -85,6 +87,8 @@ namespace cubreplication
     private:
       std::queue<stream_entry *> m_stream_entries;
 
+      std::unique_ptr<slave_control_channel> m_ctrl_chn;
```

Comment:
For me, it looks more natural that the slave_node instance is the owner of the slave_control_channel object, and the log_consumer uses a reference to it.
Generally, we may configure between several contexts to send the stream ACK : when stream position reaches the slave or when it is flushed to disk. With current design, we are required to change "owner" of the slave_control_channel either to log_consumer or stream_file (or the object responsible with flushing).
Also, from logical point of view, log_consumer handles only pure replication data, the control data seems another domain. There is the idea to change the name of log_consumer as replication_stream_consumer.

Reply:
I'd like to merge the patch and continue discussion. Please find http://jira.cubrid.org/browse/CBRD-22996